File: PAT (Basic-Level)/C/1071.c
#include <stdio.h>

int main(int argc, char const *argv[]){
    int T, K;
    scanf("%d %d", &T, &K);
    int n1, b, t, n2;
    for (int i = 0; i < K; i++){
        scanf("%d %d %d %d", &n1, &b, &t, &n2);
        if(t > T){
            printf("Not enough tokens.  Total = %d.\n", T);
        }
        else if((b == 0 && n1 > n2) || (b == 1 && n1 < n2)){
            T += t;
            printf("Win %d!  Total = %d.\n", t, T);
        }else if((b == 0 && n1 < n2) || (b == 1 && n1 > n2)){
            T -= t;
            printf("Lose %d.  Total = %d.\n", t, T);
            if(T == 0){
                printf("Game Over.\n");
                break;
            }
        }
    }
    return 0;
}

File: PAT (Basic-Level)/Python/1071.py
#!/usr/bin/python3
# -*- coding: utf-8 -*-

T, K = map(int, input().split())
for i in range(K):
    n1, b, t, n2 = map(int, input().split())
    if t > T:
        print("Not enough tokens.  Total = %d." % T)
    elif ((b == 0 and n1 > n2) or (b == 1 and n1 < n2)):
        T += t
        print("Win %d!  Total = %d." % (t, T))
    elif ((b == 0 and n1 < n2) or (b == 1 and n1 > n2)):
        T -= t
        print("Lose %d.  Total = %d." % (t, T))
        if T == 0:
            print("Game Over.")
            break 